cdap-runtime-spi/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/RuntimeJobManager.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.runtime.spi.runtimejob;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Runtime job manager to prepare and launch the job.
+ */
+public interface RuntimeJobManager {
+  /**
+   * This method uses runtime job information to launch the job.
+   *
+   * @param runtimeJobInfo runtime job information
+   * @return unique runtime job id
+   */
+  RuntimeJobId launch(RuntimeJobInfo runtimeJobInfo) throws Exception;
```

Comment:
Why stop and kill return Future while this one does not?

Reply:
If this method is async, other methods such as getDetails may not have job to get details for.